packages/gh-bin/src/cli.ts

```diff
@@ -0,0 +1,149 @@
+    console.log(HELP);
+    return;
+  }
+
+  // parse github url
+  const input = args[0];
+  const match = input.replace(/\/+$/, "").match(GITHUB_RELEASE_RE);
+  if (!match) {
+    console.error(`[ERROR] Invalid input '${input}'\n`);
+    process.exit(1);
+  }
+  let [, owner, repo, tag] = match;
+
+  if (!tag) {
+    // find a latest tag
+    // https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#get-the-latest-release
+    const latestRelease = await fetchGhApi(
+      `https://api.github.com/repos/${owner}/${repo}/releases/latest`
+    );
+    tag = latestRelease.tag_name;
+    console.log(`Found the latest release ${tag}`);
+  }
+
+  // find release assets
+  // https://docs.github.com/en/rest/releases/assets?apiVersion=2022-11-28#list-release-assets
+  const release = await fetchGhApi(
+    `https://api.github.com/repos/${owner}/${repo}/releases/tags/${tag}`
+  );
+  const assets = release.assets;
+  if (!assets || assets.length === 0) {
+    console.error(`[ERROR] No assets found for release '${tag}'\n`);
+    process.exit(1);
+  }
+
+  // prompt which files to download from assets
+  const selectedAsset = await prompts.select<string>({
+    message: "Select an asset to download",
+    options: assets.map((asset: any) => ({
+      label: asset.name,
+      value: asset.browser_download_url,
+    })),
+  });
+  if (prompts.isCancel(selectedAsset)) {
+    return;
+  }
+
+  // download a selected asset
+  const downloadSpinner = prompts.spinner();
+  downloadSpinner.start(`Downloading ${selectedAsset}`);
+  let tmpAssetPath = path.join(
+    os.tmpdir(),
+    `gh-bin-asset-${owner}-${repo}${path.extname(selectedAsset)}`
+  );
+  try {
+    const res = await fetch(selectedAsset);
+    if (!res.ok || !res.body) {
+      console.error(`[ERROR] Failed to download '${selectedAsset}'\n`);
+      process.exit(1);
+    }
+    await fs.promises.writeFile(
+      tmpAssetPath,
+      Readable.fromWeb(res.body as any)
+    );
+  } finally {
+    downloadSpinner.stop(`Downloaded ${selectedAsset}`);
+  }
+
+  // if .zip or .tar.gz, unpack and prompt again which file to use
+  if (selectedAsset.endsWith(".zip")) {
+    var zip = new AdmZip(tmpAssetPath);
+    var zipEntries = zip.getEntries();
+    const selectedEntry = await prompts.select({
+      message: "Select an entry to extract from zip",
+      options: zipEntries.map((entry) => ({
+        label: entry.name,
+        value: entry,
+      })),
+    });
+    if (prompts.isCancel(selectedEntry)) {
+      return;
+    }
+    let tmpZipEntryPath = path.join(
+      os.tmpdir(),
+      `gh-bin-asset-${owner}-${repo}-zip-${selectedEntry.name}${path.extname(selectedEntry.name)}`
+    );
+    fs.promises.writeFile(tmpZipEntryPath, selectedEntry.getData());
+    tmpAssetPath = tmpZipEntryPath;
+  } else if (selectedAsset.endsWith(".tar.gz")) {
+    console.log("[ERROR] .tar.gz unsupported (todo)");
+    process.exit(1);
```

Comment:
TODO: this is a blocker since it's common to use `.tar.gz` for linux executable release
- https://github.com/cli/cli
- https://github.com/muesli/duf
- https://github.com/lycheeverse/lychee